aws/table_aws_transfer_server.go

```diff
@@ -0,0 +1,260 @@
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getTransferServer,
+				Transform:   transform.FromField("WorkflowDetails"),
+			},
+			{
+				Name:        "structured_log_destinations",
+				Description: "Specifies the log groups to which your server logs are sent.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getTransferServer,
+				Transform:   transform.FromField("StructuredLogDestinations"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ServerId"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listTransferServers(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := TransferClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_transfer_server.listTransferServers", "client_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		return nil, nil
+	}
+	// Limiting the results
+	maxLimit := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	input := &transfer.ListServersInput{
+		MaxResults: &maxLimit,
+	}
+
+	paginator := transfer.NewListServersPaginator(svc, input, func(o *transfer.ListServersPaginatorOptions) {
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_transfer_server.listTransferServers", "api_error", err)
+			return nil, err
+		}
+
+		for _, items := range output.Servers {
+			d.StreamListItem(ctx, items)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getTransferServer(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// get service
+	svc, err := TransferClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_transfer_server.getTransferServer", "connection_error", err)
+		return nil, err
+	}
+
+	var serverID string
+	if h.Item != nil {
+		serverID = *h.Item.(types.ListedServer).ServerId
+	} else {
+		serverID = d.EqualsQuals["server_id"].GetStringValue()
+	}
+
```

Comment:
Could you please add an empty check for `serverID`?
We should enforce restrictions on API calls when a user provides an empty string in the `server_id` WHERE clause, such as (`SELECT * FROM aws_transfer_server WHERE server_id = ''`).